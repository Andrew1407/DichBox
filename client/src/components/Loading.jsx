import React from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import { componentMotion } from '../styles/motions/menu-components';
import loadingGif from '../styles/imgs/loading.gif';
import '../styles/loading.css';

const Loading = () => (
  <AnimatePresence>
  <motion.div
    { ...componentMotion }
    data-testid="loading-test"
    className="menu-form"
    id="loading-container"
  >
    <h1 id="loading-title">Loading...</h1>
    <img id="loading-img" src={ loadingGif }/>
    <span id="loading-desc">Please, stand by</span>
  </motion.div>
  </AnimatePresence>
);

export default Loading;
