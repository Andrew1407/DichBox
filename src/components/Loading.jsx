import React from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import { componentMotion } from '../styles/motions/menu-components';
import loadingGif from '../styles/imgs/loading.gif';
import '../styles/loading.css';

const Loading = () => {

  return (
    <AnimatePresence>
    <motion.div { ...componentMotion } className="menu-form">
      <h1 id="loading-title">Loading...</h1>
      <img id="loading-img" src={ loadingGif }/>
      <span id="loading-desc">Please, stand by</span>
    </motion.div>
    </AnimatePresence>
  );
};

export default Loading;
