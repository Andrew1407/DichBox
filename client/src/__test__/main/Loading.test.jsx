import React from 'react';
import { render } from '@testing-library/react';
import Loading from '../../components/Loading';
import loadingGif from '../../styles/imgs/loading.gif';

describe('Loading tests', () => {
  const { getByTestId } = render(<Loading />);
  const foundLoading = getByTestId('loading-test');

  const tests = {
    'has text content': () => {
      expect(foundLoading).toHaveTextContent('Loading...');
      expect(foundLoading).toHaveTextContent('Please, stand by');
    },
    'has image': () => {
      const loadingImg = foundLoading.querySelector('img');
      expect(loadingImg).toHaveAttribute('src', loadingGif);
    }
  };

  for (const key in tests)
    it(key, tests[key]);
});
