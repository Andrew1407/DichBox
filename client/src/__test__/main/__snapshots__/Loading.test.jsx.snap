// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading tests matches snapshot 1`] = `
<div
  className="menu-form"
  data-testid="loading-test"
  id="loading-container"
  style={
    Object {
      "transform": "translateX(-800px) translateZ(0)",
    }
  }
>
  <h1
    id="loading-title"
  >
    Loading...
  </h1>
  <img
    id="loading-img"
    src="test-file-stub"
  />
  <span
    id="loading-desc"
  >
    Please, stand by
  </span>
</div>
`;
